experimental/generation/runbot/RunBot.cs

```diff
@@ -73,14 +76,15 @@ private void LoadDialogs()
 
             Dialog lastDialog = null;
             var choices = new ChoiceSet();
+            var dialogName = _configuration["dialog"];
 
-            foreach (var resource in this.resourceExplorer.GetResources(".dialog").Where(r => r.Id.EndsWith(".main.dialog")))
+            foreach (var resource in _resourceExplorer.GetResources(".dialog").Where(r => dialogName != null ? r.Id == dialogName : true))
```

Comment:
Nit: There's a little extra work going on to compare dialogName != null for each r.
```
var dialogResources = _resourceExplorer.GetResources(".dialog");
dialogResources = dialogName != null ? dialogResources .Where(r => string.Equals(r.Id, dialogName, StringComparison.OrdinalIgnoreCase)) : dialogResources ;
foreach (var resource in dialogResources) {
```